service/src/tests/ordering.rs

```diff
@@ -0,0 +1,247 @@
+use ceramic_api::AccessModelStore;
+use ceramic_core::EventId;
+use ceramic_event::unvalidated::{
+    self,
+    signed::{self},
+    Builder,
+};
+use cid::Cid;
+use ipld_core::{ipld, ipld::Ipld};
+use recon::ReconItem;
+
+use crate::{
+    tests::{check_deliverable, gen_rand_bytes, signer},
+    CeramicEventService,
+};
+
+use super::{random_cid, random_event_id};
+
+async fn setup_service() -> CeramicEventService {
+    let _ = ceramic_metrics::init_local_tracing();
+    let conn = ceramic_store::SqlitePool::connect_in_memory()
+        .await
+        .unwrap();
+    CeramicEventService::new_without_undelivered(conn)
+        .await
+        .unwrap()
+}
+
+async fn init_event(model: Cid, signer: &signed::JwkSigner) -> signed::Event<Ipld> {
+    let init = Builder::init()
+        .with_controller("controller".to_string())
+        .with_sep("sep".to_string(), model.to_bytes())
+        .build();
+    signed::Event::from_payload(unvalidated::Payload::Init(init), signer.to_owned()).unwrap()
+}
+
+async fn data_event(
+    _model: Cid,
+    init_id: Cid,
+    prev: Cid,
+    data: Ipld,
+    signer: &signed::JwkSigner,
+) -> signed::Event<Ipld> {
+    let commit = Builder::data()
+        .with_id(init_id)
+        .with_prev(prev)
+        .with_data(data)
+        .build();
+
+    signed::Event::from_payload(unvalidated::Payload::Data(commit), signer.to_owned()).unwrap()
+}
+
+// builds init -> data -> data that are a stream (will be a different stream each call)
+async fn get_events() -> [(EventId, Vec<u8>); 3] {
+    let model = random_cid();
+    let signer = Box::new(signer().await);
+
+    let data = gen_rand_bytes::<50>();
+    let data2 = gen_rand_bytes::<50>();
+
+    let data = ipld!({
+        "radius": 1,
+        "red": 2,
+        "green": 3,
+        "blue": 4,
+        "raw": data.as_slice(),
+    });
+
+    let data2 = ipld!({
+        "radius": 1,
+        "red": 2,
+        "green": 3,
+        "blue": 4,
+        "raw": data2.as_slice(),
+    });
+
+    let init = init_event(model, &signer).await;
+    let (event_id, car) = (
+        random_event_id(Some(&init.envelope_cid().to_string())),
```

Comment:
nit, this might read a little clearer as 

```
event_id_builder().with_event(&init.envelope_cid().to_string()).build()
```

This way its clear that no random ID is actually being generated.

Reply:
Also not sure if it matters but this event ID is wrong as well. The event_id's contain the model but we do not use the model we generated above to construct the event id, instead a hard coded model is used https://github.com/ceramicnetwork/rust-ceramic/blob/50cc8baa101ba56c5c1bea31573cb0378d36c800/service/src/tests/mod.rs#L28